packages/compiler/src/core/checker.ts

```diff
@@ -2778,6 +2790,164 @@ export function createChecker(program: Program): Checker {
     return (resolved?.declarations.filter((n) => isTemplatedNode(n)) ?? []) as TemplateableNode[];
   }
 
+  function getReferencedModel(
+    propertyNode: ObjectLiteralPropertyNode | ModelPropertyNode
+  ): Model | undefined {
+    const ARRAY_MODEL_FLAG = "@@array_model_flag@@";
+
+    if (propertyNode.kind === SyntaxKind.ModelProperty) {
+      return getReferencedModelFromDecoratorArgument(propertyNode);
```

Comment:
Also missing template parameter constraint for both

Reply:
yeah I see you need a different logic when you figured out what it belongs to but I think it would be more efficient and cleaner if you have a single call to `getFirstAncestor`

Reply:
get what you mean and updated the code. please double check. thx